unittests/Threading/Thread.cpp

```diff
@@ -0,0 +1,25 @@
+//===--- Thread.cpp - Tests swift::Thread functionality -------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Threading/Thread.h"
+#include "gtest/gtest.h"
+
+using namespace swift;
+
+// Check that Thread::stackBounds() returns a reasonable value.
+TEST(StackTest, stack_bounds) {
```

Comment:
This test is definitely wrong. With the no-threads and C11 packages, it will fail, because `stackBounds()` will return without a value. I think the fix is to change the first assert into an `if (bounds.has_value()) { ... }`. The other assertions seem not unreasonable if they're inside the `if`.

Reply:
Done, thanks.